### PR TITLE
If a user has too many eth2 validators it should not longer fail with 429

### LINKED ISSUE #2719: Move ETH2 logic to a module
## Abstract

PR #2700 introduces a new module for ETH2 staking. Currently there are functions that implement the logic for eth2 but are not related to any class. The objective is to move all this logic to the new module.

## Motivation

As discussed it makes sense for eth2 staking to be represented internally as a module

### LINKED ISSUE #2835: If a user has too many ETH2 validators then the beaconcha.in stats queries return 429
## Problem Definition

We are querying beaconcha.in for the user stats by reading directly from the website since no API offers info there. Since this is not via the api, apparently if a user has too many validators (example given seems to be more than 100) then the website queries themselves return a 429 from cloudflare.

## Task Definition

To mitigate this we should:

(1) put delays in the queries and handle 429 in the stats queries
(2) eventually allow users to also import the same data from CSV (but the problem with CSV is that it needs lots of user action -- especially if they have many validators)

### PR DESCRIPTION
**Pull request by LefterisJP**
Fix #2719 
Fix #2835

### CODE REVIEW

File: rotkehlchen/chain/ethereum/modules/eth2.py [hunk 1]
```diff
@@ -23,9 +56,321 @@ def __init__(
...
+        if len(new_stats) != 0:
+            dbeth2.add_validator_daily_stats(stats=new_stats)
+
+        return dbeth2.get_validator_daily_stats(
+            validator_index=validator_index,
+            from_ts=from_timestamp,
+            to_ts=to_timestamp,
+        )
 
+    # -- Methods following the EthereumModule interface -- #
```
Comment (yabirgb) [inline rotkehlchen/chain/ethereum/modules/eth2.py:373]: At some point we should make use or get rid of those methods

**Review by yabirgb — COMMENTED** [review]
I've taken a look in the PR and everything seems fine

File: rotkehlchen/chain/ethereum/modules/eth2.py (hunk 1 above)
Comment (LefterisJP) [reply rotkehlchen/chain/ethereum/modules/eth2.py:373]: We need those for all modules. All modules should use them. Just don't wanna add too many changes in the PR.

For example eth2 should query eth2 balances for a new address from here. We already do it in the balance query, but it should go here.

**Comment by LefterisJP** [comment]
> I've taken a look in the PR and everything seems fine

Thanks a lot @yabirgb!

**Comment by LefterisJP** [comment]
The 2 failures seems to be (1) kusama node timeout and (2) coingecko 504. So seems fine and will merge.

**Pull request merged**